pkg/protocol/http1/client.go

```diff
@@ -623,7 +623,11 @@ func (c *HostClient) doNonNilReqResp(req *protocol.Request, resp *protocol.Respo
 		err = respI.ReadHeaderAndLimitBody(resp, zr, c.MaxResponseBodySize)
 	} else {
 		err = respI.ReadBodyStream(resp, zr, c.MaxResponseBodySize, func() error {
-			c.releaseConn(cc)
+			if resetConnection || req.ConnectionClose() || resp.ConnectionClose() {
```

Comment:
Looks Good!
However, I took a closer look at the `resetConnection` variable and found that there is a strange implementation above: line 562 will directly mute the logic begins with line 546-549?

Reply:
Probably not relevant, may check if you have time

Reply:
line 558 has wrote req to buffer